api/global/internal/trace.go

```diff
@@ -0,0 +1,80 @@
+package internal
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+type traceProvider struct {
+	lock    sync.Mutex
+	tracers []*tracer
+
+	delegate trace.Provider
+}
+
+var _ trace.Provider = &traceProvider{}
+
+func (p *traceProvider) setDelegate(provider trace.Provider) {
+	if p.delegate != nil {
+		return
+	}
+
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	p.delegate = provider
+	for _, t := range p.tracers {
+		t.setDelegate(provider)
+	}
+
+	p.tracers = nil
+}
+
+// Tracer creates a trace.Tracer with the given name. When a delegate is set,
+// all previously returned trace.Tracers will be swapped to equivalent
+// trace.Tracers created from the delegate.
+func (p *traceProvider) Tracer(name string) trace.Tracer {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	if p.delegate != nil {
+		return p.delegate.Tracer(name)
+	}
+
+	t := &tracer{name: name}
+	p.tracers = append(p.tracers, t)
+	return t
+}
+
+type tracer struct {
+	trace.NoopTracer
```

Comment:
I would probably leave this out of the struct and construct one as needed (since it's 0 bytes).